sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestContextManager.java

```diff
@@ -4,21 +4,38 @@
 
 import com.azure.core.test.annotation.DoNotRecord;
 
+import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
 /**
  * This class handles managing context about a test, such as custom testing annotations and verifying whether the test
  * is capable of running.
  */
+@SuppressWarnings("unchecked")
 public class TestContextManager {
+    private static final Map<String, AtomicInteger> TEST_ITERATION_CACHE = new ConcurrentHashMap<>();
```

Comment:
Do we need this TEST_ITERATION_CACHE? It is a static variable that never gets cleared. I can imagine if we have a test suite like storage this growing.
Another concern is that we use this test iteration to name our files. It assumes that the resolution of the test matrix for paramterised tests will be the same. If the parameters get resolved in a different order, our tests will break and it'll be hard to debug. "What set of parameters did foo[1].json represent?"

Reply:
Agreed that this is a bit of an issue for parameterized tests. JUnit 5 offers a variable that can be passed into `BeforeEach` method to get the parameterized iteration but it will throw an exception when used with a test that isn't parameterized. This lead me to choosing this option as the next best available route. Thinking on it more due to this comment I can look to see if the test parameters are accessible in `BeforeEach` and use them to create a hash that gets appended to the test name instead.